parlai/core/torch_ranker_agent.py

```diff
@@ -104,6 +110,34 @@ def build_model(self):
         raise NotImplementedError(
             'Abstract class: user must implement build_model()')
 
+
+    def get_train_preds(self, scores):
+        _, ranks = scores.sort(1, descending=True)
```

Comment:
Let's have a  TODO here saying we should optimize this some. The `sort` is a huge red flag, and probably the bottom for loops could be vectorized.

Reply:
oh i should move that after the if statement, we don't actually use ranks if cands=batch

Reply:
will also add todo